SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp

```diff
@@ -260,36 +274,63 @@ void USpatialSender::SendComponentUpdates(UObject* Object, const FClassInfo& Inf
 
 	UE_LOG(LogSpatialSender, Verbose, TEXT("Sending component update (object: %s, entity: %lld)"), *Object->GetName(), EntityId);
 
-	ComponentFactory UpdateFactory(Channel->GetInterestDirty(), NetDriver);
+	USpatialLatencyTracer* Tracer = USpatialLatencyTracer::GetTracer(Object);
+	ComponentFactory UpdateFactory(Channel->GetInterestDirty(), NetDriver, Tracer);
+
+	TArray<TraceKey>* TraceKeysPtr = nullptr;
+#if TRACE_LIB_ACTIVE
+	TArray<TraceKey> TraceKeys;
+	TraceKeysPtr = &TraceKeys;
+#endif
 
-	TArray<Worker_ComponentUpdate> ComponentUpdates = UpdateFactory.CreateComponentUpdates(Object, Info, EntityId, RepChanges, HandoverChanges);
+	TArray<Worker_ComponentUpdate> ComponentUpdates = UpdateFactory.CreateComponentUpdates(Object, Info, EntityId, RepChanges, HandoverChanges, TraceKeysPtr);
 
-	for (Worker_ComponentUpdate& Update : ComponentUpdates)
+	for(int i = 0; i < ComponentUpdates.Num(); i++)
 	{
+		Worker_ComponentUpdate& Update = ComponentUpdates[i];
+		TraceKey LatencyKey = USpatialLatencyTracer::InvalidTraceKey;
+#if TRACE_LIB_ACTIVE
+		checkf(TraceKeys.Num() == ComponentUpdates.Num(), TEXT("Trace keys does not match the component updates for tracing."));
+		LatencyKey = TraceKeys[i];
+#endif
 		if (!NetDriver->StaticComponentView->HasAuthority(EntityId, Update.component_id))
 		{
 			UE_LOG(LogSpatialSender, Verbose, TEXT("Trying to send component update but don't have authority! Update will be queued and sent when authority gained. Component Id: %d, entity: %lld"), Update.component_id, EntityId);
 
 			// This is a temporary fix. A task to improve this has been created: UNR-955
 			// It may be the case that upon resolving a component, we do not have authority to send the update. In this case, we queue the update, to send upon receiving authority.
 			// Note: This will break in a multi-worker context, if we try to create an entity that we don't intend to have authority over. For this reason, this fix is only temporary.
-			TArray<Worker_ComponentUpdate>& UpdatesQueuedUntilAuthority = UpdatesQueuedUntilAuthorityMap.FindOrAdd(EntityId);
-			UpdatesQueuedUntilAuthority.Add(Update);
+			FQueuedUpdate& UpdatesQueuedUntilAuthority = UpdatesQueuedUntilAuthorityMap.FindOrAdd(EntityId);
+			UpdatesQueuedUntilAuthority.ComponentUpdates.Add(Update);
```

Comment:
Can we add a TODO here with a reference to the task to squash this data into a new structure that wraps Worker_ComponentUpdate and the trace